kpops/cli/main.py

```diff
@@ -130,37 +144,51 @@ def get_step_names(steps_to_apply: list[PipelineComponent]) -> list[str]:
     return [step.name.removeprefix(step.prefix) for step in steps_to_apply]
 
 
-def filter_steps_to_apply(
-    pipeline: Pipeline, steps: set[str]
-) -> list[PipelineComponent]:
-    skipped_steps: list[str] = []
+def filter_component(
+    pipeline: Pipeline, steps: set[str], filter_type: FilterType
+) -> list:
+    filtered_steps = []
+    if filter_type == FilterType.include:
+        filtered_steps = list(
+            filter(
+                lambda component: component.name.removeprefix(component.prefix)
+                in steps,
+                pipeline,
+            )
+        )
+        log.info(f"Including the following steps: {get_step_names(filtered_steps)}")
+    elif filter_type == FilterType.exclude:
+        filtered_steps = list(
+            filter(
+                lambda component: component.name.removeprefix(component.prefix)
+                not in steps,
+                pipeline,
+            )
+        )
+        log.info(f"Excluding the following steps: {get_step_names(filtered_steps)}")
+    return filtered_steps
 
-    def filter_component(component: PipelineComponent) -> bool:
-        step_name = component.name.removeprefix(component.prefix)
-        if step_name in steps:
-            return True
-        skipped_steps.append(step_name)
-        return False
```

Comment:
I wouldn't delete the helper filter function

then you could rewrite this to avoid code duplication, e.g.
```python
    def is_in_steps(component: PipelineComponent) -> bool:
        return component.name.removeprefix(component.prefix) in steps

    filtered_steps = [
        component
        for component in pipeline
        if (
            is_in_steps(component)
            if filter_type is FilterType.include
            else not is_in_steps(component)
        )
    ]
    operation = "Including" if filter_type is FilterType.include else "Excluding"
    log.info(f"{operation} the following steps: {get_step_names(filtered_steps)}")
    return filtered_steps
```